pkg/ipcache/types.go

```diff
@@ -91,3 +127,15 @@ func (s prefixInfo) Source() source.Source {
 	}
 	return src
 }
+
+// IdentityOverride extracts any pre-determined identity from the prefix info.
+// This pre-determined identity will overwrite any identity which may be derived
+// from the prefix labels.
+func (s prefixInfo) IdentityOverride() (types.IdentityOverride, labels.Labels) {
+	for _, v := range s {
+		if v.identityOverride.IsValid() {
+			return v.identityOverride, v.labels
```

Comment:
This is probably OK in terms of resolving the conflict. IIRC `range s` won't guarantee the same item is picked if there are conflicts and the function is run multiple times. A more stable behaviour would be to pick out the conflicts, sort them, and pick an identity based on some stable heuristic.... but ideally the user will never hit conflicts because we recognize the potential for conflict ahead of time and put guard rails in place. So probably fine to just leave this for now, especially since there are warnings when conflicts are inserted.

Reply:
Made the conflict resolution deterministic.

Reply:
I added basic deterministic conflict resolution by sorting alphabetically. I'm honestly not sure what heuristic would be more useful in practice? Prefer CIDR identities over global ones over reserved ones? For two identities in the same category, which one should we pick?

Because I couldn't come up with a good rationale for anything, I decided to keep it simple for now and just sort alphabetically.